### PR TITLE
fix: `ProxyConfigArray` in type def is not an `Array`

### LINKED ISSUE #4161: ProxyConfigArray in type def (types/lib/Server.d.ts) is not an Array
### Bug report

In the bundled type delcaration file "types/lib/Server.d.ts", type `ProxyConfigArray` is defined to not an array but what its item should be.

### Actual Behavior

```ts
type ProxyConfigArray = {
  path?: string | string[] | undefined;
  context?: string | string[] | HttpProxyMiddlewareOptionsFilter | undefined;
} & HttpProxyMiddlewareOptions &
  ByPass;
```

### Expected Behavior

Below codes are modified base on `@types/webpack-dev-server` codes:
```ts
type ProxyConfigArrayItem = {
    path?: string | string[] | undefined;
    context?: string | string[] | HttpProxyMiddlewareOptionsFilter | undefined;
} & HttpProxyMiddlewareOptions &
  ByPass;

type ProxyConfigArray = ProxyConfigArrayItem[];
```

### How Do We Reproduce?

```ts
import type { Configuration } from 'webpack-dev-server'
const apiPrefixes: string[] = ['/api/']
const devServerOptions: Configuration = {
  proxy: [
    {
      // TS2322: Type '{ context: string[]; target: string; }' is not assignable to type 'Options'.
      // Object literal may only specify known properties, and 'context' does not exist in type 'Options'.
      context: apiPrefixes,
      target : '',
    },
  ],
}
```

### Please paste the results of `npx webpack-cli info` here, and mention other relevant information
Not using webpack-cli.
using webpack-dev-server@4.7.2
using typescript@4.5.4

### PR DESCRIPTION
**Pull request by whitetrefoil**
- [x] This is a **bugfix**
- [ ] This is a **feature**
- [ ] This is a **code refactor**
- [ ] This is a **test update**
- [ ] This is a **docs update**
- [ ] This is a **metadata update**

### For Bugs and Features; did you add new tests?

No, it's a pure type declaration fix, no actual code.

### Motivation / Use-Case

Fixes #4161

### Breaking Changes

No breaking change.

### CODE REVIEW

**Review by alexander-akait — APPROVED** [review]

File: lib/Server.js [hunk 1]
```diff
@@ -121,11 +121,15 @@ const schema = require("./options.json");
  * @callback ByPass
  * @param {Request} req
  * @param {Response} res
- * @param {ProxyConfigArray} proxyConfig
+ * @param {ProxyConfigArrayItem} proxyConfig
```
Comment (alexander-akait) [inline lib/Server.js:124]: I think something wrong here `ByPass` can't have `ByPass` inside

Comment (whitetrefoil) [reply lib/Server.js:124]: I haven't actually run the code yet, but looked at line 2222 it seemed to just pass the whole `proxyConfig` as-is:
```js
const isByPassFuncDefined =
  typeof proxyConfig.bypass === "function";
const bypassUrl = isByPassFuncDefined
  ? await proxyConfig.bypass(req, res, proxyConfig)
  : null;
```

Comment (alexander-akait) [reply lib/Server.js:124]: Can you provide configuration where we have a problem with types?

Comment (whitetrefoil) [reply lib/Server.js:124]: Sure~ I've put a minimial repro. in the original issue #4161:
```ts
import type { Configuration } from 'webpack-dev-server'
const devServerOptions: Configuration = {
  proxy: [
    {
      // TS2322: Type '{ context: never[]; target: string; }' is not assignable to type 'Options'.
      // Object literal may only specify known properties, and 'context' does not exist in type 'Options'.
      context: [],
      target : '',
    },
  ],
}
```
That code works when using the old `@types/webpack-dev-server`, but throws TS error when using `webpack-dev-server`'s new bundle types.  (And it actually works if I just ignores the TS type error)

Comment (alexander-akait) [reply lib/Server.js:124]: We have more errors - try https://github.com/webpack/webpack-dev-server/pull/4173

Comment (whitetrefoil) [reply lib/Server.js:124]: #4173 works for my case, should I close this one?

Comment (alexander-akait) [reply lib/Server.js:124]: Yep, let's close, merged https://github.com/webpack/webpack-dev-server/pull/4173, anyway thanks for PR

**Pull request closed without merging**